apps/web/pagesAndRewritePaths.js

```diff
@@ -27,11 +23,26 @@ let subdomainRegExp = (exports.subdomainRegExp = getSubdomainRegExp(
 ));
 exports.orgHostPath = `^(?<orgSlug>${subdomainRegExp})\\.(?!vercel\.app).*`;
 
-let beforeRewriteExcludePages = pages.concat(otherNonExistingRoutePrefixes);
-exports.orgUserRoutePath = `/:user((?!${beforeRewriteExcludePages.join("|")}|_next|public)[a-zA-Z0-9\-_]+)`;
-exports.orgUserTypeRoutePath = `/:user((?!${beforeRewriteExcludePages.join(
-  "/|"
-)}|_next/|public/)[^/]+)/:type((?!avatar\.png)[^/]+)`;
-exports.orgUserTypeEmbedRoutePath = `/:user((?!${beforeRewriteExcludePages.join(
-  "/|"
-)}|_next/|public/)[^/]+)/:type/embed`;
+/**
+ * Returns a regex that matches all existing routes, virtual routes (like /forms, /router, /success etc) and nextjs special paths (_next, public)
+ */
+function getRegExpMatchingAllReservedRoutes(suffix) {
+  // Following routes don't exist but they work by doing rewrite. Thus they need to be excluded from matching the orgRewrite patterns
+  // Make sure to keep it upto date as more nonExistingRouteRewrites are added.
+  const otherNonExistingRoutePrefixes = ["forms", "router", "success", "cancel"];
+  const nextJsSpecialPaths = ["_next", "public"];
+
+  let beforeRewriteExcludePages = pages.concat(otherNonExistingRoutePrefixes).concat(nextJsSpecialPaths);
```

Comment:
+ suffix ensures that the last item is followed by suffix but not a pipe symbol.